sql/src/main/java/io/crate/executor/transport/TransportShardUpsertAction.java

```diff
@@ -194,10 +194,12 @@ public TransportShardUpsertAction(Settings settings,
             try {
                 location = shardIndexOperationOnReplica(request, item, indexShard);
             } catch (Exception e) {
-                Exceptions.rethrowUnchecked(e);
+                if (!TransportActions.isShardNotAvailableException(e)) {
```

Comment:
afaik exception should also not be thrown on VersionConflictEngineException. if unsure add at least a FIXME